File: backend/src/main/java/com/internship/management/service/DashboardService.java
package com.internship.management.service;

import com.internship.management.dto.response.AdminDashboardResponse;
import com.internship.management.dto.response.InternDashboardResponse;

public interface DashboardService {
    AdminDashboardResponse getAdminDashboard();
    InternDashboardResponse getInternDashboard();
}

File: frontend/src/services/dashboardService.js
import api from '../api/axios';

const dashboardService = {
  /**
   * Fetch Administrator Dashboard statistics and recent activities.
   * @returns {Promise<Object>} ApiResponse wrapping AdminDashboardResponse
   */
  getAdminDashboard: async () => {
    const response = await api.get('/api/dashboard/admin');
    return response.data;
  },

  /**
   * Fetch Intern Dashboard workspace statistics and logs.
   * @returns {Promise<Object>} ApiResponse wrapping InternDashboardResponse
   */
  getInternDashboard: async () => {
    const response = await api.get('/api/dashboard/intern');
    return response.data;
  }
};

export default dashboardService;
